support-lambdas/stripe-intent/cfn.yaml

```diff
@@ -70,84 +59,29 @@ Resources:
-    Type: "AWS::ApiGateway::BasePathMapping"
-    Properties:
-      RestApiId: !Ref ServerlessRestApi # auto generated by the Transform
-      DomainName: !Ref DomainName
-      Stage: Stage
-    DependsOn: ServerlessRestApiProdStage # auto generated by the Transform
-
-  DNSRecord:
-    Type: AWS::Route53::RecordSet
-    Properties:
-      HostedZoneName: support.guardianapis.com.
-      Name: !FindInMap [ StageMap, !Ref Stage, DomainName ]
-      Comment: !Sub CNAME for strip-intent API ${Stage}
-      Type: CNAME
-      TTL: '120'
-      ResourceRecords:
-        - !GetAtt [ DomainName, RegionalDomainName ]
-
-  4XXAlarm:
-    Type: AWS::CloudWatch::Alarm
-    Properties:
-      AlarmActions:
-        - !Sub arn:aws:sns:${AWS::Region}:${AWS::AccountId}:reader-revenue-dev
-      AlarmName: !Sub High 4XX rate for support-${Stage}-stripe-intent
-      MetricName: 4XXError
-      Namespace: AWS/ApiGateway
-      Dimensions:
-        - Name: ApiName
-          Value: !Sub support-${Stage}-stripe-intent
-        - Name: Stage
-          Value: !Ref Stage
-        - Name: Method
-          Value: POST
-        - Name: Resource
-          Value: /stripe-intent
-      ComparisonOperator: GreaterThanOrEqualToThreshold
-      Threshold: 1
-      Period: 60
-      EvaluationPeriods: 1
-      Statistic: Sum
-
-  5XXAlarm:
-    Type: AWS::CloudWatch::Alarm
-    Properties:
-      AlarmActions:
-        - !Sub arn:aws:sns:${AWS::Region}:${AWS::AccountId}:reader-revenue-dev
-      AlarmName: !Sub High 5XX rate for support-${Stage}-stripe-intent
-      MetricName: 5XXError
-      Namespace: AWS/ApiGateway
-      Dimensions:
-        - Name: ApiName
-          Value: !Sub support-${Stage}-stripe-intent
-        - Name: Stage
-          Value: !Ref Stage
-        - Name: Method
-          Value: POST
-        - Name: Resource
-          Value: /stripe-intent
-      ComparisonOperator: GreaterThanOrEqualToThreshold
-      Threshold: 1
-      Period: 60
-      EvaluationPeriods: 1
-      Statistic: Sum
+  StripeIntentAlarm:
+      Type: AWS::CloudWatch::Alarm
+      Condition: ProdOnlyResource
+      Properties:
+        AlarmActions:
+          - !Sub arn:aws:sns:${AWS::Region}:${AWS::AccountId}:reader-revenue-dev
+        AlarmName: !Join
+            - ' '
+            - - !FindInMap [ Constants, Alarm, Urgent ]
+              - 'PROD'
+              - 'Stripe create setup intent Lambda failing'
+        AlarmDescription: !Join
+            - ' '
+            - - 'Impact - Credit card checkout will not work for subscriptions or recurring contributions.'
+              - !FindInMap [ Constants, Alarm, Process ]
+        Namespace: AWS/Lambda
+        MetricName: Errors
+        Dimensions:
+          - Name: FunctionName
+            Value: !Ref LambdaFunctionOverHttps
+        Statistic: Sum
+        ComparisonOperator: GreaterThanThreshold
+        Threshold: 0
+        EvaluationPeriods: 1
+        Period: 60
+        TreatMissingData: "breaching"
```

Comment:
missing data wouldn't be breaching, it would just be noone has caused an error/called the lambda I think